### PR TITLE
docs: add developer target for building docs

### LINKED ISSUE #762: docs: add developer makefile target for building docs
The command for building HTML docs is simple but non-trivial. We should add a Makefile target for doing so, such that developers can iterate locally and confirm that the docs are linted.

### PR DESCRIPTION
**Pull request by austin3dickey**
To address @jonkeane 's thoughts in https://github.com/conbench/conbench/pull/761#issue-1592417932, this PR adds a Makefile target and instructions for iterating locally with docs. Closes #762 .

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -75,6 +75,13 @@ rebuild-expected-api-docs: run-app-bg
 	rm _new_api_docs.json
 	git diff ./conbench/tests/api/_expected_docs.py
 
+
+# Build HTML docs, exiting non-zero if there are warnings. Those can be very bad.
```
Comment (jonkeane) [inline Makefile]: While I agree that erroring on warnings in CI builds | deploys is good, not sure we want that when iterating locally. Could we make that an option?

File: README.md [hunk 2]
```diff
@@ -115,6 +115,18 @@ Deploys the Conbench API server to a local minikube-powered Kubernetes cluster.
...
+* `make docs-build`: Builds HTML docs locally so you may check that they render
+  correctly with no linting problems. You may install dependencies with
+  ```
+  pip install \
+      -r requirements-dev.txt \
+      -e ./benchclients/python \
+      -e ./benchalerts \
+      -e ./benchadapt/python \
+      -e ./benchconnect \
+      -e ./benchrun/python \
```
Comment (jonkeane) [inline README.md]: We can keep this here, but we should probably also mention that installing _all the_ dependencies (eg benchalerts, etc.) isn't totally necessary if one isn't touching that. 

If I were contributing to the philosophy + general docs having to have functional installs of all of those is a pretty big hurdle to clear that isn't in principle important for us (right now at least).

File: Makefile (hunk 1 above)
Comment (austin3dickey) [reply Makefile]: Done: removed it from the Makefile command since one can just look at the warnings locally

File: README.md (hunk 2 above)
Comment (austin3dickey) [reply README.md]: Yeah great point. Clarified!

File: README.md [hunk 3]
```diff
@@ -115,6 +115,15 @@ Deploys the Conbench API server to a local minikube-powered Kubernetes cluster.
...
 
+* `make docs-build`: Builds HTML docs locally so you may check that they render
+  correctly with no linting problems. Dependencies can be installed with
+  `pip install -r requirements-dev.txt`. Also, if you're working on the docstrings of
+  any of this repo's python packages, ensure the package is installed locally before
+  using this command.
+
+  In CI, we use `make build-docs SPHINXOPTS='-W --keep-going'` to fail the build if
+  there are Sphinx warnings. When using this command locally, you can just do
+  `make build-docs`, but keep an eye on the warnings.
```
Comment (jgehrcke) [inline README.md:126]: :heart:

**Pull request merged**